scripts/backend.py

```diff
@@ -28,6 +41,42 @@
             "swing":      {"port": 6602, "client": None},
             "cafe":       {"port": 6603, "client": None}}
 
+# needed for cookie generation
+app.config["SECRET_KEY"] = "YouShouldProbablyChangeThis"
+
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view = 'login'
+
+DEFAULT_LIVESTREAM_INFO = {
+    'active': False,
+    'current_dj': None,
+    'last_played': [],
+    'PORT': 6601
+}
+
+# namedtuple definition has to happen before loading
+LivestreamTrack = namedtuple('LivestreamTrack', ['artist', 'title', 'first_seen'])
+
+LIVESTREAM_INFO = db.load_pickle(DEFAULT_LIVESTREAM_INFO)
+
+
+# be safe with our savedata
+def handle_shutdown(*args):
+    db.save_pickle(LIVESTREAM_INFO)
+    exit(1)
+
+
+def handle_imminent_shutdown(*args):
+    pickle_path = os.path.join(db.SAVEDATA_PATH, 'livestream_save.pickle')
+    os.remove(pickle_path)  # something went terribly wrong so better to get rid of our savedata
+    exit(2)
+
+
+signal.signal(signal.SIGRTMIN, handle_imminent_shutdown)
```

Comment:
I've never caught `SIGRTMIN` before, is there any point in doing so?  How common is it?